### DISCUSSION TITLE
ArgoCD multi-region / DR

### DISCUSSION THREAD

**Discussion by tatroc** [argoproj/argo-cd#9488] [Q&A] [answered]
How is the ArgoCD community handling DR/HA with ArgoCD. For example, if I have an application that runs in 1 region primary and I want to plan for a region failover. If ArgoCD is in us-east-1 with the application and us-east-1 goes down. Are ArgoCD operators syncing the config to another ArgoCD instance ?

**Comment by jgwest** [comment]
AFAIK, Argo CD does not have a mechanism for syncing config between multiple Argo CD instances, or multiple Argo CD clusters. (This would be an interesting 3rd party project, but I'm not personally aware of anything like this).

For this scenario, you would need to ensure that a subset of yourK8s cluster master nodes kept going in this scenario, which would be a requirement of your K8s cloud provider/operator. For example, might want to use a K8s cluster than has master nodes that span multiple regions (https://kubernetes.io/docs/setup/best-practices/multiple-zones/).

For reference:
- https://argo-cd.readthedocs.io/en/stable/operator-manual/disaster_recovery/
- https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/

**Comment by rumstead** [comment, marked as answer]
I am also very interested in what folks are doing here. Here is where we have landed.

We use the App of Apps pattern for deploying our Argo CD Applications and Application Sets. The TL;DR is we use a central repository to control the Argo CD Applications/sets via linting, change management tickets, and "wave releases".

We run Argo CD HA, in two different regions and deploy the Argo CD Applications/sets to both regions in a HOT-HOT manner. To avoid the two controllers from fighting with each other, we use sync windows to block the syncing of all the resources that the secondary region would be deploying. 

It isn't the cheapest way of doing things but allows us to failover our "ADMIN" Argo CD cluster very fast since by enabling and disabling a sync window.

**Reply by nastacio** [reply]
> We run Argo CD HA, in two different regions and deploy the Argo CD Applications/sets to both regions in a HOT-HOT manner. To avoid the two controllers from fighting with each other, we use sync windows to block the syncing of all the resources that the secondary region would be deploying.

Not a bad way to go. I was thinking of the same HOT-HOT approach and hadn't though of the sync windows to avoid clashes. Pretty clever.

**Reply by rumstead** [reply]
Took a few failed iterations to get there 🤣

**Reply by tatroc** [reply]
Would love to see a blog post on this :).
